lfsapi/client.go

```diff
@@ -275,7 +284,7 @@ func newRequestForRetry(req *http.Request, location string) (*http.Request, erro
 
 	for key := range req.Header {
 		if key == "Authorization" {
-			if req.URL.Scheme != newReq.URL.Scheme || req.URL.Host != newReq.URL.Host {
```

Comment:
That sounds sensible to me. I wonder if there could ever be a legitimate use case for a https ==> http redirect?!